### ISSUE TITLE
bundle install fails

### ISSUE THREAD

**Issue by spotlesscoder** [3ofcoins/chef-browser#68]
Unfortunately, I couldn't install the dependencies.

Any idea how I can fix it? I'm not familiar with ruby.

Logs:

```
user@user-VirtualBox:~/chef-browser$ bundle install
Fetching gem metadata from https://rubygems.org/.............
Using rake 10.4.2
Using addressable 2.3.8
Using buff-extensions 1.0.0
Using hashie 3.4.2
Using varia_model 0.4.1
Using buff-config 1.0.1
Using buff-ignore 1.1.1
Using buff-ruby_engine 0.1.0
Using buff-shell_out 0.2.0
Using builder 3.2.2
Using bundler 1.16.1
Using mime-types 1.25.1
Using mini_portile 0.6.2
Using nokogiri 1.6.6.2
Using rack 1.6.4
Using rack-test 0.6.3
Using xpath 2.0.0
Using capybara 2.5.0
Using hitimes 1.2.3
Using timers 4.0.4
Using celluloid 0.16.0
Using nio4r 1.1.1
Using celluloid-io 0.16.2
Fetching charlock_holmes 0.7.3
Installing charlock_holmes 0.7.3 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/charlock_holmes-0.7.3/ext/charlock_holmes
/home/user/.rbenv/versions/2.5.0/bin/ruby -r ./siteconf20180411-6489-1gy7stu.rb
extconf.rb
checking for -licui18n... no
checking for -licui18n... no


***************************************************************************************
*********** icu required (brew install icu4c or apt-get install libicu-dev)
***********
***************************************************************************************
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/home/user/.rbenv/versions/2.5.0/bin/$(RUBY_BASE_NAME)
	--with-icu-dir
	--without-icu-dir
	--with-icu-include
	--without-icu-include=${icu-dir}/include
	--with-icu-lib
	--without-icu-lib=${icu-dir}/lib
	--with-icui18nlib
	--without-icui18nlib
	--with-icui18nlib
	--without-icui18nlib

To see why this extension failed to compile, please check the mkmf.log which can
be found here:

/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0-static/charlock_holmes-0.7.3/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in
/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/charlock_holmes-0.7.3
for inspection.
Results logged to
/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0-static/charlock_holmes-0.7.3/gem_make.out

An error occurred while installing charlock_holmes (0.7.3), and Bundler
cannot continue.
Make sure that `gem install charlock_holmes -v '0.7.3'` succeeds before
bundling.

In Gemfile:
  github-linguist was resolved to 3.5.2, which depends on
    charlock_holmes

```

```
user@user-VirtualBox:~/chef-browser$ cat /home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0-static/charlock_holmes-0.7.3/gem_make.out
current directory: /home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/charlock_holmes-0.7.3/ext/charlock_holmes
/home/user/.rbenv/versions/2.5.0/bin/ruby -r ./siteconf20180411-6489-1gy7stu.rb extconf.rb
checking for -licui18n... no
checking for -licui18n... no


***************************************************************************************
*********** icu required (brew install icu4c or apt-get install libicu-dev) ***********
***************************************************************************************
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/home/user/.rbenv/versions/2.5.0/bin/$(RUBY_BASE_NAME)
	--with-icu-dir
	--without-icu-dir
	--with-icu-include
	--without-icu-include=${icu-dir}/include
	--with-icu-lib
	--without-icu-lib=${icu-dir}/lib
	--with-icui18nlib
	--without-icui18nlib
	--with-icui18nlib
	--without-icui18nlib

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0-static/charlock_holmes-0.7.3/mkmf.log

extconf failed, exit code 1
user@user-VirtualBox:~/chef-browser$ cat /home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0-static/charlock_holmes-0.7.3/mkmf.log
have_library: checking for -licui18n... -------------------- no

"gcc -o conftest -I/home/user/.rbenv/versions/2.5.0/include/ruby-2.5.0/x86_64-linux -I/home/user/.rbenv/versions/2.5.0/include/ruby-2.5.0/ruby/backward -I/home/user/.rbenv/versions/2.5.0/include/ruby-2.5.0 -I. -I/home/user/.rbenv/versions/2.5.0/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wno-maybe-uninitialized conftest.c  -L. -L/home/user/.rbenv/versions/2.5.0/lib -Wl,-rpath,/home/user/.rbenv/versions/2.5.0/lib -L. -L/home/user/.rbenv/versions/2.5.0/lib  -fstack-protector -rdynamic -Wl,-export-dynamic     -Wl,-rpath,/home/user/.rbenv/versions/2.5.0/lib -L/home/user/.rbenv/versions/2.5.0/lib -lruby-static  -lpthread -ldl -lcrypt -lm   -lc"
checked program was:
/* begin */
1: #include "ruby.h"
2: 
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

"gcc -o conftest -I/home/user/.rbenv/versions/2.5.0/include/ruby-2.5.0/x86_64-linux -I/home/user/.rbenv/versions/2.5.0/include/ruby-2.5.0/ruby/backward -I/home/user/.rbenv/versions/2.5.0/include/ruby-2.5.0 -I. -I/home/user/.rbenv/versions/2.5.0/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wno-maybe-uninitialized conftest.c  -L. -L/home/user/.rbenv/versions/2.5.0/lib -Wl,-rpath,/home/user/.rbenv/versions/2.5.0/lib -L. -L/home/user/.rbenv/versions/2.5.0/lib  -fstack-protector -rdynamic -Wl,-export-dynamic     -Wl,-rpath,/home/user/.rbenv/versions/2.5.0/lib -L/home/user/.rbenv/versions/2.5.0/lib -lruby-static -licui18n  -lpthread -ldl -lcrypt -lm   -lc"
/usr/bin/ld: cannot find -licui18n
collect2: error: ld returned 1 exit status
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: 
14: int t(void) { ; return 0; }
/* end */

--------------------

have_library: checking for -licui18n... -------------------- no

"gcc -o conftest -I/home/user/.rbenv/versions/2.5.0/include/ruby-2.5.0/x86_64-linux -I/home/user/.rbenv/versions/2.5.0/include/ruby-2.5.0/ruby/backward -I/home/user/.rbenv/versions/2.5.0/include/ruby-2.5.0 -I. -I/home/user/.rbenv/versions/2.5.0/include    -O3 -ggdb3 -Wall -Wextra -Wno-unused-parameter -Wno-parentheses -Wno-long-long -Wno-missing-field-initializers -Wno-tautological-compare -Wno-parentheses-equality -Wno-constant-logical-operand -Wno-self-assign -Wunused-variable -Wimplicit-int -Wpointer-arith -Wwrite-strings -Wdeclaration-after-statement -Wimplicit-function-declaration -Wdeprecated-declarations -Wno-packed-bitfield-compat -Wsuggest-attribute=noreturn -Wsuggest-attribute=format -Wno-maybe-uninitialized conftest.c  -L. -L/home/user/.rbenv/versions/2.5.0/lib -Wl,-rpath,/home/user/.rbenv/versions/2.5.0/lib -L. -L/home/user/.rbenv/versions/2.5.0/lib  -fstack-protector -rdynamic -Wl,-export-dynamic     -Wl,-rpath,/home/user/.rbenv/versions/2.5.0/lib -L/home/user/.rbenv/versions/2.5.0/lib -lruby-static -licui18n  -lpthread -ldl -lcrypt -lm   -lc"
/usr/bin/ld: cannot find -licui18n
collect2: error: ld returned 1 exit status
checked program was:
/* begin */
 1: #include "ruby.h"
 2: 
 3: /*top*/
 4: extern int t(void);
 5: int main(int argc, char **argv)
 6: {
 7:   if (argc > 1000000) {
 8:     printf("%p", &t);
 9:   }
10: 
11:   return 0;
12: }
13: 
14: int t(void) { ; return 0; }
/* end */

--------------------


```

Thanks already!

**Comment by spotlesscoder** [comment]
installed the missing package: apt-get install libicu-dev

bundle install still failing:

```

user@user-VirtualBox:~/chef-browser$ bundle install
Fetching gem metadata from https://rubygems.org/.............
Using rake 10.4.2
Using addressable 2.3.8
Using buff-extensions 1.0.0
Using hashie 3.4.2
Using varia_model 0.4.1
Using buff-config 1.0.1
Using buff-ignore 1.1.1
Using buff-ruby_engine 0.1.0
Using buff-shell_out 0.2.0
Using builder 3.2.2
Using bundler 1.16.1
Using mime-types 1.25.1
Using mini_portile 0.6.2
Using nokogiri 1.6.6.2
Using rack 1.6.4
Using rack-test 0.6.3
Using xpath 2.0.0
Using capybara 2.5.0
Using hitimes 1.2.3
Using timers 4.0.4
Using celluloid 0.16.0
Using nio4r 1.1.1
Using celluloid-io 0.16.2
Using charlock_holmes 0.7.3
Using libyajl2 1.2.0
Fetching ffi-yajl 2.2.2
Installing ffi-yajl 2.2.2 with native extensions
Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

current directory:
/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/ffi-yajl-2.2.2/ext/ffi_yajl/ext/encoder
/home/user/.rbenv/versions/2.5.0/bin/ruby -r ./siteconf20180411-8594-1jfne4y.rb
extconf.rb
-I/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/libyajl2-1.2.0/lib/libyajl2/vendored-libyajl2/include
$(cflags)
-L/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/libyajl2-1.2.0/lib/libyajl2/vendored-libyajl2/lib
-L. -L/home/user/.rbenv/versions/2.5.0/lib  -fstack-protector -rdynamic
-Wl,-export-dynamic
checking for yajl/yajl_tree.h... yes
creating Makefile

current directory:
/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/ffi-yajl-2.2.2/ext/ffi_yajl/ext/encoder
make "DESTDIR=" clean

current directory:
/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/ffi-yajl-2.2.2/ext/ffi_yajl/ext/encoder
make "DESTDIR="
compiling encoder.c
encoder.c: In function ‘gen_number’:
encoder.c:109:3: warning: ISO C90 forbids mixed declarations and code
[-Wdeclaration-after-statement]
   char *cptr = RSTRING_PTR(str);
   ^
encoder.c: In function ‘rb_cNilClass_ffi_yajl’:
encoder.c:262:30: warning: passing argument 2 of ‘gen_cstring’ discards ‘const’
qualifier from pointer target type [-Wdiscarded-qualifiers]
     gen_cstring(rb_yajl_gen, "", sizeof("")-1);
                              ^
encoder.c:79:7: note: expected ‘char *’ but argument is of type ‘const char *’
 VALUE gen_cstring(VALUE rb_yajl_gen, char *cptr, int len) {
       ^
encoder.c: In function ‘rb_cTrueClass_ffi_yajl’:
encoder.c:272:30: warning: passing argument 2 of ‘gen_cstring’ discards ‘const’
qualifier from pointer target type [-Wdiscarded-qualifiers]
     gen_cstring(rb_yajl_gen, "true", sizeof("true")-1);
                              ^
encoder.c:79:7: note: expected ‘char *’ but argument is of type ‘const char *’
 VALUE gen_cstring(VALUE rb_yajl_gen, char *cptr, int len) {
       ^
encoder.c: In function ‘rb_cFalseClass_ffi_yajl’:
encoder.c:282:30: warning: passing argument 2 of ‘gen_cstring’ discards ‘const’
qualifier from pointer target type [-Wdiscarded-qualifiers]
     gen_cstring(rb_yajl_gen, "false", sizeof("false")-1);
                              ^
encoder.c:79:7: note: expected ‘char *’ but argument is of type ‘const char *’
 VALUE gen_cstring(VALUE rb_yajl_gen, char *cptr, int len) {
       ^
encoder.c: In function ‘Init_encoder’:
encoder.c:366:3: warning: ISO C90 forbids mixed declarations and code
[-Wdeclaration-after-statement]
   ID sym_Date = rb_intern("Date");
   ^
encoder.c:368:3: warning: ISO C90 forbids mixed declarations and code
[-Wdeclaration-after-statement]
   ID sym_Time = rb_intern("Time");
   ^
encoder.c:370:3: warning: ISO C90 forbids mixed declarations and code
[-Wdeclaration-after-statement]
   ID sym_DateTime = rb_intern("DateTime");
   ^
encoder.c:372:3: warning: ISO C90 forbids mixed declarations and code
[-Wdeclaration-after-statement]
   ID sym_StringIO = rb_intern("StringIO");
   ^
encoder.c:380:20: error: ‘rb_cFixnum’ undeclared (first use in this function)
   rb_define_method(rb_cFixnum, "ffi_yajl", rb_cFixnum_ffi_yajl, 2);
                    ^
encoder.c:380:20: note: each undeclared identifier is reported only once for
each function it appears in
encoder.c:381:20: error: ‘rb_cBignum’ undeclared (first use in this function)
   rb_define_method(rb_cBignum, "ffi_yajl", rb_cBignum_ffi_yajl, 2);
                    ^
encoder.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-self-assign’
cc1: warning: unrecognized command line option ‘-Wno-constant-logical-operand’
cc1: warning: unrecognized command line option ‘-Wno-parentheses-equality’
cc1: warning: unrecognized command line option ‘-Wno-tautological-compare’
Makefile:241: recipe for target 'encoder.o' failed
make: *** [encoder.o] Error 1

make failed, exit code 2

Gem files will remain installed in
/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/ffi-yajl-2.2.2 for
inspection.
Results logged to
/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0-static/ffi-yajl-2.2.2/gem_make.out

An error occurred while installing ffi-yajl (2.2.2), and Bundler cannot
continue.
Make sure that `gem install ffi-yajl -v '2.2.2'` succeeds before bundling.

In Gemfile:
  chef-zero was resolved to 4.3.2, which depends on
    ffi-yajl

```

```
user@user-VirtualBox:~/chef-browser$ cat /home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/extensions/x86_64-linux/2.5.0-static/ffi-yajl-2.2.2/gem_make.out
current directory: /home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/ffi-yajl-2.2.2/ext/ffi_yajl/ext/encoder
/home/user/.rbenv/versions/2.5.0/bin/ruby -r ./siteconf20180411-11583-1nik0k1.rb extconf.rb
 -I/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/libyajl2-1.2.0/lib/libyajl2/vendored-libyajl2/include $(cflags)
 -L/home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/libyajl2-1.2.0/lib/libyajl2/vendored-libyajl2/lib -L. -L/home/user/.rbenv/versions/2.5.0/lib  -fstack-protector -rdynamic -Wl,-export-dynamic
checking for yajl/yajl_tree.h... yes
creating Makefile

current directory: /home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/ffi-yajl-2.2.2/ext/ffi_yajl/ext/encoder
make "DESTDIR=" clean

current directory: /home/user/.rbenv/versions/2.5.0/lib/ruby/gems/2.5.0/gems/ffi-yajl-2.2.2/ext/ffi_yajl/ext/encoder
make "DESTDIR="
compiling encoder.c
encoder.c: In function ‘gen_number’:
encoder.c:109:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   char *cptr = RSTRING_PTR(str);
   ^
encoder.c: In function ‘rb_cNilClass_ffi_yajl’:
encoder.c:262:30: warning: passing argument 2 of ‘gen_cstring’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
     gen_cstring(rb_yajl_gen, "", sizeof("")-1);
                              ^
encoder.c:79:7: note: expected ‘char *’ but argument is of type ‘const char *’
 VALUE gen_cstring(VALUE rb_yajl_gen, char *cptr, int len) {
       ^
encoder.c: In function ‘rb_cTrueClass_ffi_yajl’:
encoder.c:272:30: warning: passing argument 2 of ‘gen_cstring’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
     gen_cstring(rb_yajl_gen, "true", sizeof("true")-1);
                              ^
encoder.c:79:7: note: expected ‘char *’ but argument is of type ‘const char *’
 VALUE gen_cstring(VALUE rb_yajl_gen, char *cptr, int len) {
       ^
encoder.c: In function ‘rb_cFalseClass_ffi_yajl’:
encoder.c:282:30: warning: passing argument 2 of ‘gen_cstring’ discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
     gen_cstring(rb_yajl_gen, "false", sizeof("false")-1);
                              ^
encoder.c:79:7: note: expected ‘char *’ but argument is of type ‘const char *’
 VALUE gen_cstring(VALUE rb_yajl_gen, char *cptr, int len) {
       ^
encoder.c: In function ‘Init_encoder’:
encoder.c:366:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   ID sym_Date = rb_intern("Date");
   ^
encoder.c:368:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   ID sym_Time = rb_intern("Time");
   ^
encoder.c:370:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   ID sym_DateTime = rb_intern("DateTime");
   ^
encoder.c:372:3: warning: ISO C90 forbids mixed declarations and code [-Wdeclaration-after-statement]
   ID sym_StringIO = rb_intern("StringIO");
   ^
encoder.c:380:20: error: ‘rb_cFixnum’ undeclared (first use in this function)
   rb_define_method(rb_cFixnum, "ffi_yajl", rb_cFixnum_ffi_yajl, 2);
                    ^
encoder.c:380:20: note: each undeclared identifier is reported only once for each function it appears in
encoder.c:381:20: error: ‘rb_cBignum’ undeclared (first use in this function)
   rb_define_method(rb_cBignum, "ffi_yajl", rb_cBignum_ffi_yajl, 2);
                    ^
encoder.c: At top level:
cc1: warning: unrecognized command line option ‘-Wno-self-assign’
cc1: warning: unrecognized command line option ‘-Wno-constant-logical-operand’
cc1: warning: unrecognized command line option ‘-Wno-parentheses-equality’
cc1: warning: unrecognized command line option ‘-Wno-tautological-compare’
Makefile:241: recipe for target 'encoder.o' failed
make: *** [encoder.o] Error 1

make failed, exit code 2

```

**Comment by spotlesscoder** [comment]
I don't understand why chef-zero gets resolved to 4.3.2 as there is already a version 14.x out there

**Comment by xamebax** [comment]
Hey, thanks for reporting this issue. This project isn't under active development right now and a number of the gems it was using are outdated. I will merge an outstanding PR that updates some of the gems later today, and look into your issue.

**Comment by spotlesscoder** [comment]
Great to hear that you still care about the project.

Thanks in advance!

**Comment by xamebax** [comment]
Updating the dependencies uncovered some bugs so everything took longer than I anticipated, apologies.

The problem here was that Chef Browser didn't support Ruby 2.5, which you were using. This is now fixed with the 2.0.0 release. Could you check if using the newest release fixes the installation issue?

Thanks for the note about libicu-dev, I'll add it to the README.

(As per your question about chef-zero, they actually skipped quite a few numbers (6-12) in their [versioning](https://github.com/chef/chef-zero/blob/master/CHANGELOG.md); I'm guessing to sync with chef-client.)

**Comment by spotlesscoder** [comment]
Great, thanks
I also had to install cmake and pkg-config
